[#41] Migrate to Spring Data

Sheptytskyid opened the pull request.

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

GetFire committed:
[#4],[#6] REFACTOR

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	.travis.yml
#	pom.xml
#	sonar-project.properties
#	src/main/java/net/greatstart/model/Contact.java

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
implemented Spring Data

Commit:
in process of configuring connection to DB and rewriting unit tests

Commit:
in process of configuring connection to DB and rewriting unit tests

Commit:
working with old DataBaseConfig

Commit:
fixed unit tests

Commit:
program works but tests fail

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into spring_data

# Conflicts:
#	src/main/java/net/greatstart/controllers/UserController.java

Sheptytskyid commented:
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

Sheptytskyid commented on src/main/java/net/greatstart/configs/SecurityConfig.java:
```diff
@@ -12,19 +12,24 @@
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
     private UserService userService;
+    private DataSource dataSource;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "dataSource" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into spring_data

# Conflicts:
#	src/main/java/net/greatstart/controllers/UserController.java

Commit:
fixed unit tests
excluded model package from coveralls

Commit:
fixed DataSource schema in properties

Commit:
removed unused field in SecurityConfig

holkwen commented:
У меня новые проекты перестали добавляться.

holkwen commented:
Лечится добавлением в pom.xml:
        &lt;dependency&gt;
            &lt;groupId&gt;org.hibernate&lt;/groupId&gt;
            &lt;artifactId&gt;hibernate-java8&lt;/artifactId&gt;
            &lt;version&gt;${hibernate.version}&lt;/version&gt;
        &lt;/dependency&gt;

Commit:
removed unused field in SecurityConfig

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into spring_data

# Conflicts:
#	src/main/java/net/greatstart/dao/hibernate/InvestmentInterestDaoImpl.java

Commit:
updated unit tests added InvInterestControllerTest

Commit:
added GreatStartUserDetailsServiceTest

Commit:
added GreatStartUserDetailsServiceTest

Commit:
updated UserConverterServiceTest

Commit:
excluded dto package from coverage report

Commit:
added UserValidationServiceTest

Commit:
updated LoginCOntrollerTest

Commit:
fixed badges in readme.md
added PowerMock dependency in pom.xml
added processRegistration test to RegistrationControllerTest

Commit:
added SecurityServiceTest

Commit:
added SecurityServiceTest

zozich commented on pom.xml:
```diff
@@ -130,6 +142,12 @@
                 <groupId>org.jacoco</groupId>
                 <artifactId>jacoco-maven-plugin</artifactId>
                 <version>0.7.8</version>
+                <configuration>
```
This is not needed. Your `model` and `dto` classes should be tested withing testing other layers

zozich commented on pom.xml:
```diff
@@ -87,10 +82,27 @@
             <scope>test</scope>
         </dependency>
         <dependency>
+            <groupId>org.powermock</groupId>
```
Remove if not needed

zozich commented on src/main/java/net/greatstart/dao/ProjectDao.java:
```diff
@@ -1,11 +1,15 @@
+public interface ProjectDao extends CrudRepository<Project, Long> {
 
-    List<Project> getByUserId(long id);
+    List<Project> findAll(Pageable numberOfRecords);
```
Don't you have it by default?

zozich commented on src/main/java/net/greatstart/dto/DtoUser.java:
```diff
@@ -1,12 +1,14 @@
 import javax.validation.constraints.Size;
 
 @Data
+@Component
```
Why?

zozich commented on src/main/java/net/greatstart/dto/DtoUserProfile.java:
```diff
@@ -1,11 +1,13 @@
 import javax.validation.constraints.Size;
 
 @Data
+@Component
```
Why? Remove everywhere

zozich commented on src/main/java/net/greatstart/controllers/EventController.java:
```diff
@@ -17,16 +17,12 @@ public EventController(EventService eventService) {
-    public boolean createEvent(Event event) {
-        return eventService.createEvent(event);
+    public void saveEvent(Event event) {
+        eventService.saveEvent(event);
```
save should return saved entity as done by default by Spring Data

zozich reviewed (changes requested):
Please amend

Commit:
Service methods "save" now return objects
UserValidationService's method validate is now not static
updated unit tests

Commit:
Merge branch 'master' of https://github.com/Sheptytskyid/JavaEnterpriseFinalProjectGroup6 into spring_data

Commit:
removed powermock dependency from pom.xml

Commit:
removed configs package and Main class from jacoco's report

Commit:
removed Main class from jacoco's report

Commit:
all dao methods now extend PagingAndSortingRepository to enable pageable requests

Commit:
fixec unit tests

Commit:
removed redundant Component annotations in model layer

Commit:
fixed AbstractModel

zozich reviewed (approved):
Okay now
